ci/cloudbuild/build.sh

```diff
@@ -94,33 +94,43 @@ if [[ "${LOCAL_BUILD}" = "true" ]]; then
   io::log_h1 "Starting local build: ${BUILD_NAME}"
   readonly TIMEFORMAT="==> 🕑 ${BUILD_NAME} completed in %R seconds"
   time "${PROGRAM_DIR}/builds/${BUILD_NAME}.sh"
-elif [[ -n "${DISTRO}" ]]; then
-  # Quick checks to surface invalid arguments early
-  if [ ! -r "ci/cloudbuild/${DISTRO}.Dockerfile" ]; then
-    echo "Unknown distro: ${DISTRO}"
-    print_usage
-    exit 1
-  elif [ ! -x "ci/cloudbuild/builds/${BUILD_NAME}.sh" ]; then
-    echo "Unknown build name: ${BUILD_NAME}"
+  exit
+fi
+
+# If --distro wasn't specified, look it up from the build's trigger file.
+if [[ -z "${DISTRO}" ]]; then
+  trigger_file="ci/cloudbuild/triggers/${BUILD_NAME}.yaml"
+  DISTRO="$(grep _DISTRO "${trigger_file}" | awk '{print $2}' || true)"
```

Comment:
nit: (maybe for a future PR) we should consider something like https://kislyuk.github.io/yq/ instead of parsing the YAML file with grep.

Reply:
Thanks. I did consider `yq` already, but it's a dep that most of us likely don't have on our machines, and it's not just one `apt install` away. If we ever need to do non-trivial parsing of yaml, then I think it'd be worth using something like `yq`, but until then, I suspect this is probably OK.... well, that was my thinking anyway :man_shrugging: :smile: